### ISSUE TITLE
A binding to CopperSpice

### ISSUE THREAD

**Issue by aferust** [tim-dlang/dqt#3]
https://forum.dlang.org/post/ybsmhfpvpsiqepbgmjif@forum.dlang.org

Can your binding generator be used for a d binding to [copperspice](https://www.copperspice.com/)? Recently I saw this on the forum. It looks like CopperSpice is a better alternative to QT.

Thanks

**Comment by tim-dlang** [comment]
It could be possible, but there are some differences between Qt and CopperSpice, which could make it more difficult.

One advantage of Qt is, that it does not use many types from the C++ standard library in the headers and instead implements custom types like QList and QString. The D bindings then only need to support these Qt types which are the same on all platforms, like Windows and Linux. CopperSpice seems to use more types from the C++ standard library. QList is for example implemented using std::deque: https://github.com/copperspice/copperspice/blob/07be95f61d3dce519fae209517f145b629f6e67a/src/core/containers/qlist.h#L444
Different implementation of the C++ standard can implement types differently, so the bindings would need to be different for different C++ standard libraries and versions.

**Comment by aferust** [comment]
I see.  I still think that binding should be easier than porting the whole thing.  I ported a c++ project to dlang a year ago. I had some time, and it took about two weeks of mine. I had to find d equivalents of c++ std. I remember it was not too difficult, but eventually, I realized it was doable.

https://github.com/aferust/epanet-d-dev --> my port
https://github.com/OpenWaterAnalytics/epanet-dev --> the original c++ project.

I thought I could handle creating a copperspice binding, but I have never heavily used Qt. It would be better someone familiar to handle this. I know that those things require motivation and time.

Thank you for answering. I am closing this issue now.
